report-generator/src/main/java/com/teamscale/report/testwise/model/factory/TestInfoFactory.java

```diff
@@ -65,11 +65,20 @@ public TestInfo createFor(TestCoverageBuilder testCoverageBuilder) {
 	public List<TestInfo> createTestInfosWithoutCoverage() {
 		ArrayList<TestInfo> results = new ArrayList<>();
 		for (TestDetails testDetails : testDetailsMap.values()) {
-			if (uniformPathsWithCoverage.contains(testDetails.uniformPath)) {
+			if (!uniformPathsWithCoverage.contains(testDetails.uniformPath)) {
 				TestInfoBuilder testInfo = new TestInfoBuilder(testDetails.uniformPath);
 				testInfo.setDetails(testDetails);
 				testInfo.setExecution(testExecutionsMap.get(testDetails.uniformPath));
 				results.add(testInfo.build());
+				uniformPathsWithCoverage.add(testDetails.uniformPath);
+			}
+		}
+		for (TestExecution testExecution : testExecutionsMap.values()) {
```

Comment:
Why do we need this second loop? Shouldn't all entries in the `testExecutionsMap` have coverage and are thus not relevant for the test infos **without** coverage?

Reply:
You are right! I just thought this would make it more robust if you do not provide all relevant artifacts, but testwise coverage with executed tests and no coverage is probably always a misconfiguration. Instead of adding the info I now print an error for those cases.